admin/client.go

```diff
@@ -691,9 +691,15 @@ func (e *GenericOpenAPIError) SetError(errorString string) {
 	e.error = errorString
 }
 
-// format error message using title and detail when model implements Error
-func formatErrorMessage(status, path, method string, v ApiError) string {
-	return fmt.Sprintf("%v %v: HTTP %v (Error code: %q) Detail: %v Reason: %v. Params: %v",
+// FormatErrorMessageWithDetails: format error message using error struct fields: Should be only used for testing
+func FormatErrorMessageWithDetails(status, path, method string, v ApiError) string {
+	detailString := ""
+	if v.BadRequestDetail != nil {
+		badRequestDetail, _ := json.Marshal(v.GetBadRequestDetail())
+		detailString = string(badRequestDetail)
```

Comment:
```suggestion
		if badRequestDetail, err := json.Marshal(v.GetBadRequestDetail()); err != nil {
		   detailString = string(badRequestDetail)
		}
```

shouldn't we check for errors converting JSON?

Reply:
We do parse JSON first from backend response. Even if we had invalid JSON  at this point somehow by regression in future would you see us doing different handling here? TL:DR - SDK does have request logger that shows rawa data. I do not like to throw errors from error handling method